recipes/repo.rb

```diff
@@ -284,6 +284,102 @@
+end
+
+package 'createrepo-agent'
+
+package 'createrepo-c'
+
+package 'rpm'
+
+package 'socat'
+
+node['ros_buildfarm']['rpm_repos'].each do |dist, versions|
+  dist_dir = "/var/repos/#{dist}"
+
+  %w(building testing main).each do |repo|
+    repo_dir = "#{dist_dir}/#{repo}"
+    file "#{repo_dir}/RPM-GPG-KEY-ros-#{repo}" do
+      action :delete
+    end
+
+    versions.each do |version, architectures|
+      version_dir = "#{repo_dir}/#{version}"
+      srpms_dir = "#{version_dir}/SRPMS"
+      [dist_dir, repo_dir, version_dir, srpms_dir].each do |dir|
+        directory dir do
+          owner agent_username
+          group agent_username
+          mode '0755'
+        end
+      end
+
+      execute "createrepo_c --no-database #{srpms_dir}" do
+        user agent_username
+        group agent_username
+        not_if { ::File.exist?("#{srpms_dir}/repodata/repomd.xml") }
+      end
+
+      execute "gpg --armor --detach --sign --yes --default-key=#{gpg_key['fingerprint']} #{srpms_dir}/repodata/repomd.xml" do
+        user agent_username
+        group agent_username
+        environment 'HOME' => "/home/#{agent_username}"
+        not_if { ::File.exist?("#{srpms_dir}/repodata/repomd.xml.asc") }
+      end
+
+      architectures.each do |arch|
+        arch_dir = "#{version_dir}/#{arch}"
+        debug_dir = "#{arch_dir}/debug"
+        [arch_dir, debug_dir].each do |dir|
+          directory dir do
+            owner agent_username
+            group agent_username
+            mode '0755'
+          end
+        end
+
+        execute "createrepo_c --no-database #{arch_dir} --excludes=debug/*" do
+          user agent_username
+          group agent_username
+          not_if { ::File.exist?("#{arch_dir}/repodata/repomd.xml") }
+        end
+
+        execute "gpg --armor --detach --sign --yes --default-key=#{gpg_key['fingerprint']} #{arch_dir}/repodata/repomd.xml" do
+          user agent_username
+          group agent_username
+          environment 'HOME' => "/home/#{agent_username}"
+          not_if { ::File.exist?("#{arch_dir}/repodata/repomd.xml.asc") }
+        end
+
+        execute "createrepo_c --no-database #{debug_dir}" do
+          user agent_username
+          group agent_username
+          not_if { ::File.exist?("#{debug_dir}/repodata/repomd.xml") }
+        end
+
+        execute "gpg --armor --detach --sign --yes --default-key=#{gpg_key['fingerprint']} #{debug_dir}/repodata/repomd.xml" do
+          user agent_username
+          group agent_username
+          environment 'HOME' => "/home/#{agent_username}"
+          not_if { ::File.exist?("#{debug_dir}/repodata/repomd.xml.asc") }
+        end
+      end
+    end
+  end
+end
+
+if not node['ros_buildfarm']['rpm_bootstrap_url'].empty?
+  file "/home/#{agent_username}/ros_bootstrap_rpm_urls.txt" do
+    owner agent_username
+    group agent_username
+    mode '0644'
+    content node['ros_buildfarm']['rpm_bootstrap_url']
+  end
+end
```

Comment:
Since the filename is plural should we make the attribute match plurality as list by default?

```suggestion
if not node['ros_buildfarm']['rpm_bootstrap_urls'].empty?
  file "/home/#{agent_username}/ros_bootstrap_rpm_urls.txt" do
    owner agent_username
    group agent_username
    mode '0644'
    content node['ros_buildfarm']['rpm_bootstrap_urls'].join("\n")
  end
end
```